pkg/littlefs2/fs/littlefs2_fs.c

```diff
@@ -117,11 +124,21 @@ static int prepare(littlefs2_desc_t *fs)
 
     memset(&fs->fs, 0, sizeof(fs->fs));
 
-    if (!fs->config.block_count) {
-        fs->config.block_count = fs->dev->sector_count - fs->base_addr;
+    size_t block_size = fs->dev->pages_per_sector * fs->dev->page_size;
+#if CONFIG_LITTLEFS2_MIN_BLOCK_SIZE_EXP >= 0
+    if (!(block_size = ((1u << CONFIG_LITTLEFS2_MIN_BLOCK_SIZE_EXP) / block_size) * block_size)) {
+        block_size = fs->dev->pages_per_sector * fs->dev->page_size;
     }
+#endif
+    size_t block_count = fs->dev->sector_count * fs->dev->pages_per_sector * fs->dev->page_size;
+    block_count /= block_size;
+    fs->sectors_per_block = block_size / (fs->dev->pages_per_sector * fs->dev->page_size);
```

Comment:
To avoid overflow for MTD devices > 4 GiB

```suggestion
    fs->sectors_per_block = block_size / (fs->dev->pages_per_sector * fs->dev->page_size);
    size_t block_count = fs->dev->sector_count / fs->sectors_per_block;
```

Reply:
fixed